Ghetto Seed Extraction with tools

Ms-Mee opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This Pull Request implements a way to extract seeds from produce with tools, time and sound effects may need changing, however the current state is usable and does not seem to have issues.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Given the sheer emphasis on ghetto analogues for processes and how boring placing produce in a seed extractor to get seeds is, I see this as a nice addition that makes ghetto botany slightly more accessible.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Added a means of extracting seeds using sharp objects and tools. As well as a chance for composting produce to plant the same type of produce on an empty tray.
balance: New ghetto options for seed extraction and planting unreliant on autolathes.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Ms-Mee committed:
Initial form, probably broken

triplezeta commented:
yes

Ms-Mee committed:
Just fixing the base

mcmeiler commented on code/modules/hydroponics/grown.dm:
```diff
@@ -85,6 +85,18 @@
+		playsound(loc, 'sound/weapons/slice.ogg', 50, TRUE, -1)
+		user.visible_message("<span class='notice'>[user] starts slicing apart \the [src].</span>", "<span class='notice'>You start slicing apart \the [src]...</span>", "<span class='hear'>You hear the sound of a sharp object slicing some plant matter.</span>")
+		if(do_after(user, 50, target = src))
+			to_chat(user, "<span class='notice'>You slice apart this [src].</span>")
```
```suggestion
			to_chat(user, "<span class='notice'>You slice apart [src].</span>")
```
Pretty sure ``src`` already contains a "the"
![grafik](https://user-images.githubusercontent.com/83819203/135280199-4ae3ac60-1d54-4b71-ae5e-f6fd03bdcfb7.png)

Ms-Mee replied:
Yeah I'll change it once I have the other tools in place and all, not quite as it's intended just there so I don't forget what's what

Ms-Mee committed:
More tool options and kinda better text

All for today, most likely, gotta look into supposedly it dropping no seeds when done in-hand and making the process have steps per plant, later.

mcmeiler commented on code/modules/hydroponics/grown.dm:
```diff
@@ -85,6 +85,49 @@
+			/*	new /obj/item/reagent_containers/food/snacks/grown/chipped(user.drop_location(), 1)
+				use(1) */
+				seedify(src, 1)
+		if(W.tool_behaviour == TOOL_WIRECUTTER)
```
```suggestion
		if(TOOL_WIRECUTTER)

mcmeiler replied:
Unnecessary for ``switch()``

mcmeiler commented on code/modules/hydroponics/grown.dm:
```diff
@@ -85,6 +85,49 @@
+			if(do_after(user, 50, target = src))
+				to_chat(user, "<span class='notice'>You dig into the [src] to collect it's seeds! It's all gross and unusuable now, ew!</span>")
+				seedify(src, 1)
+		if(W.tool_behaviour == TOOL_CROWBAR)
```
```suggestion
		if(TOOL_CROWBAR)

mcmeiler commented on code/modules/hydroponics/grown.dm:
```diff
@@ -85,6 +85,49 @@
+			/*	new /obj/item/reagent_containers/food/snacks/grown/split(user.drop_location(), 1)
+				use(1) */
+				seedify(src, 1)
+		if(W.tool_behaviour == TOOL_WRENCH)
```
```suggestion
		if(TOOL_WRENCH)

mcmeiler replied:
Keep in mind what that you may have to do manual fixing if you commit these changes.

These changes fixed compiling for me in VSC, and they still work in-game.

mcmeiler commented on code/modules/hydroponics/grown.dm:
```diff
@@ -85,6 +85,49 @@
+				to_chat(user, "<span class='notice'>You slice apart the [src]! You went too far and the tiny remaining scraps are worthless!</span>")
+			/*	new /obj/item/reagent_containers/food/snacks/grown/sliced(user.drop_location(), 1)
+				use(1) */
+				seedify(src, 1)
```
```suggestion
	if(W.get_sharpness())
		playsound(loc, 'sound/weapons/slice.ogg', 50, TRUE, -1)
		user.visible_message("<span class='notice'>[user] starts slicing apart \the [src].</span>", "<span class='notice'>You start slicing apart \the [src]...</span>", "<span class='hear'>You hear the sound of a sharp object slicing some plant matter.</span>")
		if(do_after(user, 50, target = src))
			to_chat(user, "<span class='notice'>You slice apart the [src]! You went too far and the tiny remaining scraps are worthless!</span>")
		/*	new /obj/item/reagent_containers/food/snacks/grown/sliced(user.drop_location(), 1)
			use(1) */
			seedify(src, 1)
	switch(W.tool_behaviour)

mcmeiler replied:
Removing the faulty ``switch()`` and putting the right one at the bottom, also fixing indentation.

mcmeiler commented on code/modules/hydroponics/grown.dm:
```diff
@@ -85,6 +85,49 @@
+			/*	new /obj/item/reagent_containers/food/snacks/grown/sliced(user.drop_location(), 1)
+				use(1) */
+				seedify(src, 1)
+		if(W.tool_behaviour == TOOL_SCREWDRIVER)
```
```suggestion
		if(TOOL_SCREWDRIVER)

Ms-Mee committed:
Fixing switch dummy, no runtimes now time to test locally what's up with the hand thing, also adds hemostat as an option

Ms-Mee committed:
Stinky switch being stinky, so maybe this?

Ms-Mee committed:
WHY WAS IT GIVING COMPILE ERRORS BEFORE AAAAAA

Ms-Mee committed:
MOAIS FLAMED AHAHAHA

Ms-Mee commented:
A deal has been struck with a particularly twisted easter deity, now all that is left is the leftover items and adjusting the time and sounds, which is optional.
Unless checks fails, I hope such a fate comes not

Ms-Mee committed:
GET SQUASHED NERD
(Secret tool that will help us later)

Ms-Mee committed:
BULLSHIT TIMING LESS GO

Ms-Mee committed:
Cancelled feature

MarkSuckerberg commented on code/modules/hydroponics/grown.dm:
```diff
@@ -85,6 +85,71 @@
 			for(var/datum/plant_gene/trait/T in seed.genes)
 				T.on_attackby(src, O, user)
 
+/proc/flamingmoai(obj/item/O, t_max, mob/living/user, var/d_item)
```
🗿

Ms-Mee committed:
COMPOSTED MOAI

Ms-Mee committed:
1/4 chance to make it a little less reliable

Ms-Mee committed:
Merge branch 'shiptest-ss13:master' into SeedExtract

Ms-Mee committed:
Goodbye, my hero

Ms-Mee committed:
You will not be forgotten

Ms-Mee committed:
Damn

Ms-Mee committed:
Fuck you all my homies ghetto their seeds

Ms-Mee committed:
Surely this must be for jest?

Ms-Mee committed:
I may be fucking stupid

Ms-Mee committed:
You win, this time, dm

Ms-Mee committed:
But just this once!

Ms-Mee committed:
...

Ms-Mee commented:
Sure hope this does not break, it should not, if it does, I WILL scream. You have been warned.
